### ISSUE TITLE
Can't install identity service in image for Ubuntu 20.04

### ISSUE THREAD

**Issue by davidguidali** [Azure/iotedge#6853]
I'm trying to install the aziot identity service (and aziot edged) into a docker image based on Ubuntu 20.04. 
Thats how my Dockerfile looks like:

```
FROM ubuntu:20.04
RUN apt-get update && apt-get install -y \
	curl \
	apt-transport-https \
	software-properties-common

RUN groupadd docker

RUN curl -L https://github.com/Azure/azure-iotedge/releases/download/1.4.3/aziot-identity-service_1.4.1-1_ubuntu20.04_amd64.deb -o aziot-identity-service.deb && dpkg -i ./aziot-identity-service.deb 

RUN curl -L https://github.com/Azure/azure-iotedge/releases/download/1.4.3/aziot-edge_1.4.3-1_ubuntu20.04_amd64.deb -o aziot-edged.deb && dpkg -i ./aziot-edged.deb
```

When trying to build the image, I get this error:

```
 > [4/8] RUN curl -L https://github.com/Azure/azure-iotedge/releases/download/1.4.3/aziot-identity-service_1.4.1-1_ubuntu20.04_amd64.deb -o aziot-identity-service.deb && dpkg -i ./aziot-identity-service.deb:
#7 0.382   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#7 0.382                                  Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 3772k  100 3772k    0     0  6596k      0 --:--:-- --:--:-- --:--:-- 6596k
#7 0.986 Selecting previously unselected package aziot-identity-service.
#7 0.990 (Reading database ... 10818 files and directories currently installed.)
#7 0.990 Preparing to unpack ./aziot-identity-service.deb ...
#7 1.151 Unpacking aziot-identity-service (1.4.1-1) ...
#7 1.525 dpkg: dependency problems prevent configuration of aziot-identity-service:
#7 1.525  aziot-identity-service depends on libtss2-esys0 (>= 2.3.1); however:
#7 1.525   Package libtss2-esys0 is not installed.
#7 1.525
#7 1.525 dpkg: error processing package aziot-identity-service (--install):
#7 1.525  dependency problems - leaving unconfigured
#7 1.534 Errors were encountered while processing:
#7 1.534  aziot-identity-service
```

As a workaround i use the package for Ubuntu 18.04, which works flawlessly:
`RUN curl -L https://github.com/Azure/azure-iotedge/releases/download/1.4.3/aziot-identity-service_1.4.1-1_ubuntu18.04_amd64.deb -o aziot-identity-service.deb && dpkg -i ./aziot-identity-service.deb `

Just now I'm wondering if its even possible to run iotedge inside a docker containr, since there will be no systemd. Does iotedge rely on systemd to be available?

**Comment by vipeller** [comment]
hi @davidguidali , no iotedged does not require systemd

**Comment by davidguidali** [comment]
> 

Thanks for your reply. However, If i try to configure & run iotedge within a docker container, i get the following error message:

```
Restarting service for configuration to take effect...

Stopping atziot-edged.service...System has not been booted with systemd as init system (PID 1). Can't operate. 
Failed to connect to bus: Host is down
```

This error is thrown for each service (identity service, key service etc.) of the iotedge runtime.

**Comment by micahl** [comment]
@davidguidali Running the identity service (or iot edge) inside a container is not a scenario we test or support.  The individual services do not have a hard dependency on systemd.  Some of the CLI commands, however, do assume the presence of systemd.

What's the motivation for needing to run the identity service + iot edge in a container?

**Comment by davidguidali** [comment]
> @davidguidali Running the identity service (or iot edge) inside a container is not a scenario we test or support. The individual services do not have a hard dependency on systemd. Some of the CLI commands, however, do assume the presence of systemd.
> 
> What's the motivation for needing to run the identity service + iot edge in a container?

Thanks for your answer, I will then try manage the services w/o systemd inside a container.
We have build a setup for integration tests for out edge modules where we used to run everything in containers to allow better parallel test execution and scaling. It used to work pretty well and now we try to migrate from 1.1 to 1.4.

**Comment by micahl** [comment]
@davidguidali got it.  If you haven't already, take a look at these [dev docs](https://github.com/Azure/iot-identity-service/tree/main/docs-dev) in the identity service repo (if you haven't already), specifically the [Running the services locally](https://github.com/Azure/iot-identity-service/blob/main/docs-dev/running/index.md).  And for the aziot-edged take a look at the [Run section ](https://github.com/Azure/iotedge/blob/main/edgelet/doc/devguide.md#run) in its respective dev doc page.

If that answers your question then let's close out this issue.

**Comment by davidguidali** [comment]
> @davidguidali got it. If you haven't already, take a look at these [dev docs](https://github.com/Azure/iot-identity-service/tree/main/docs-dev) in the identity service repo, specifically the [Running the services locally](https://github.com/Azure/iot-identity-service/blob/main/docs-dev/running/index.md). And for the aziot-edged take a look at the [Run section ](https://github.com/Azure/iotedge/blob/main/edgelet/doc/devguide.md#run) in its respective dev doc page.
> 
> If that answers your question then let's close out this issue.

Thanks for your answer. I'll give it a look. But I would keep the issue open since at least installing on the Ubuntu 20.04 image should work anyway, shouldn't it?

**Comment by micahl** [comment]
Installing (not in a container) on 20.04 works.  Since we don’t test or support the scenario of running in a container there’s no guarantee (or expectation) that it should work there, so no.  

That being said, package dependencies for IoT Edge can vary between distro releases and dpkg doesn’t auto resolve them.  In this case 18.04 doesn’t have a package for the TSS libraries so we build and include what is required in our package, but since an official package is available with 20.04 our package relies on it as a dependency. When installing IoT Edge on a Debian-based system use apt-get.

**Comment by davidguidali** [comment]
Alrighth thanks for your information. Feel free to close the issue. :)
